py-polars/polars/internals/dataframe/frame.py

```diff
@@ -315,8 +317,30 @@ def estimated_size(self) -> int:
         this function returns the visible size of the buffer, not its total capacity.
 
         FFI buffers are included in this estimation.
+
+        Parameters
+        ----------
+        unit : str, optional
+            Scale returned size to the given unit ("b", "kb", "mb", "gb", or "tb").
```

Comment:
Could you include the choices in the parameter type description? See the conventions [here](https://numpydoc.readthedocs.io/en/latest/format.html#parameters).

```suggestion
        unit : {None, 'b', 'kb', 'mb', 'gb', 'tb'}
            Scale returned size to the given unit.
```

Reply:
Note: `unit` is optional to supply (when calling) as it defaults to "b" - but can't be None (will move the units though).

> Nice utility, but shouldn't we try to implement these things on the Rust side? I'd rather keep our Python bindings as skinny as possible.

Perhaps, but given it could return int/float depending on the supplied unit, and it's the sort of thing you are only likely to interactively eyeball with different units, seemed like more of a burden to add it down there.

Reply:
Updated docstring 👍